src/goose/toolkit/web_browser.py

```diff
@@ -0,0 +1,442 @@
+import sys
+import importlib.util
+import os
+import subprocess
+import time
+import random
+import shutil
+from typing import Callable
+
+# Windows-specific import
+if sys.platform.startswith("win"):
+    import winreg
+
+# Check and install selenium if not installed
+if importlib.util.find_spec("selenium") is None:
+    subprocess.check_call(["python", "-m", "pip", "install", "selenium"])
```

Comment:
just wondering what happens if this fails (vpn/internal registry). does it end up in a crash loop in which case you have to SIGINT/SIGKILL goose? i don't quite remember if we can recover but we should check

Reply:
I believe it will just crash